src/transaction/boot_sr.c

```diff
@@ -1197,63 +1188,22 @@ boot_find_rest_temp_volumes (THREAD_ENTRY * thread_p, VOLID volid,
     }
   temp_name = fileio_get_base_file_name (boot_Db_full_name);
 
-  if (boot_Db_parm->temp_nvols > 0)
+  /* Cycle over all temporarily volumes, skip the given one */
+  for (num_vols = boot_Db_parm->temp_last_volid; num_vols <= LOG_MAX_DBVOLID; num_vols++)
     {
-      /* Cycle over all temporarily volumes, skip the given one */
-      if (forward_dir)
+      temp_volid = (VOLID) num_vols;
+      if (temp_volid != NULL_VOLID)
```

Comment:
이 조건은 for 루프에서 매번 검사할 필요가 없습니다. for 루프 이전에 1회만 검사하는게 좋겠습니다.

Reply:
```suggestion
      if (boot_Db_parm->temp_last_volid == NULL_VOLID)
        {
          num_vols = 0;
        }
      else
        {
          num_vols = boot_Db_parm->temp_last_volid;
        }
      for(; num_vols <= LOG_MAX_DBVOLID; num_vols++)
        {
          ...
        }
```

해당 조건은 NULL_VOLID인 -1의 케이스를 건너뛰려는 의도로 보입니다. 제안하신 for 루프 이전에 1회 검사하려면 위 코드로 적용해야 합니다. 위 flow를 말씀하신게 맞을까요?

Reply:
구도로 답변드렸습니다. boot_Db_parm->temp_last_volid가 NULL_VOLID인 경우에 대해서는 이 함수의 역할이 아닙니다. 이 함수를 호출해 주는 코드에서 주석 처리한 부분을 풀어주세요.

추가적으로, VOLID temp_volid; 로컬 변수도 제거할 수 있지 않을까요?
1. VOLID temp_volid 제거
2. for loop에 사용되는 int num_vols -> int temp_volid로 네이밍 변경
-- 기존 num_vols 사용부 모두 temp_volid로 변경
3. temp_volid를 함수 인자로 넘길 때 (VOLID) 형변환

Reply:
제 생각엔 num_vols를 제거하고 VOLID 타입인 temp_volid로 대체하는게 가독성 면에서 더 좋을 것 같습니다.
1번에서 말씀하신게 num_vols 지우는 것을 말씀하신게 맞으실까요?

Reply:
이 쓰레드에서 요청드렸던 내용 반영해 주세요.

Reply:
반영했습니다.